### PR TITLE
added eng stemming logic, extracted stemming from main flow 

### PR DESCRIPTION
**Pull request by gevzoran**
created new stemming module, and added unit tests

### CODE REVIEW

File: apputils/test/test_stemming_eng_logic.py [hunk 1]
```diff
@@ -0,0 +1,76 @@
+import logging
+from unittest import TestCase
+
+from stemming import stemming
+
+logging.basicConfig(level = logging.INFO)
+logger = logging.getLogger('stemmingTest')
+
+
```
Comment (smartFreshBear) [inline apputils/test/test_stemming_eng_logic.py:9]: for next time,each commit need to have 1 agenda.
you pushed 3 agenda with 1 commit

Comment (gevzoran) [reply apputils/test/test_stemming_eng_logic.py]: will keep in mind next time

File: main_flow/main_flow.py [hunk 2]
```diff
@@ -57,24 +58,14 @@ def loadCache():
...
 def periodically_save_cache():
     while True:
         # Wait for new data to accumulate
         time.sleep(CYCLE_TIME_TO_SAVE_CACHE)
 
         # Save to disk
-        with open('data.pkl', 'wb') as handle:
-            pickle.dump(from_word_to_steam_cache, handle, protocol=pickle.HIGHEST_PROTOCOL)
+        with open('../stemming/data.pkl', 'wb') as handle:
+            pickle.dump(from_word_to_stem_cache, handle, protocol=pickle.HIGHEST_PROTOCOL)
```
Comment (smartFreshBear) [inline main_flow/main_flow.py:68]: I think this will break the linux server

Comment (gevzoran) [reply main_flow/main_flow.py:68]: i moved data.pkl from mainflow to stemming and this path changed automatically with refactor
i will look into this..

Comment (gevzoran) [reply main_flow/main_flow.py:68]: why would it break the server? is that path format for windows?>

Comment (smartFreshBear) [reply main_flow/main_flow.py:68]: ok sorry I see you moved the file
but please check out this method
def periodically_save_cache():
    while True:
        # Wait for new data to accumulate
        time.sleep(CYCLE_TIME_TO_SAVE_CACHE)

        # Save to disk
        with open('data.pkl', 'wb') as handle:
            pickle.dump(from_word_to_steam_cache, handle, protocol=pickle.HIGHEST_PROTOCOL)

make sure we are saving it in the new location

Comment (smartFreshBear) [reply main_flow/main_flow.py:68]: ohh ok my mistake...
BUT 
why this method is still in main_flow?
it's related only to stem - c0an you please open a ticket to move it to stemming?(or do it as a new commit?

**Pull request merged**